sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -186,26 +190,90 @@ private SdkFunctionMetadata CreateSdkFunctionMetadata(string functionName, strin
 
             return bindingMetadata;
         }
 
-        private void AddOutputBindingsAndExtensions(IList<ExpandoObject> bindingMetadata, MethodDefinition method)
+        private void AddOutputBindingsAndExtensions(ModuleDefinition module, IList<ExpandoObject> bindingMetadata, MethodDefinition method)
         {
+            if (!TryAddOutputBindingFromMethod(bindingMetadata, method))
+            {
+                AddOutputBindingsFromReturnType(module, bindingMetadata, method);
+            }
+        }
+
+        private void AddOutputBindingsFromReturnType(ModuleDefinition module, IList<ExpandoObject> bindingMetadata, MethodDefinition method)
+        {
+            string? returnType = GetActualDataTypeOrNull(method.ReturnType);
+
+            if (returnType is not null)
+            {
+                if (returnType == HttpResponseType)
+                {
+                    AddHttpOutputBinding(bindingMetadata);
+                }
+                else
+                {
+                    TypeDefinition returnDefinition = module.GetType(returnType)
+                        ?? throw new InvalidOperationException($"Couldn't find the type definition {returnType}");
+
+                    foreach (PropertyDefinition property in returnDefinition.Properties)
+                    {
+                        AddOutputBindingFromProperty(bindingMetadata, property);
+                    }
+                }
+            }
+        }
+
+        private void AddOutputBindingFromProperty(IList<ExpandoObject> bindingMetadata, PropertyDefinition property)
+        {
+            foreach (CustomAttribute propertyAttribute in property.CustomAttributes)
+            {
+                if (IsOutputBindingType(propertyAttribute))
+                {
+                    AddOutputBindingMetadata(bindingMetadata, propertyAttribute, property.Name);
+                    AddExtensionInfo(_extensions, propertyAttribute);
+
+                    // Only 1 output binding per property
+                    return;
+                }
+            }
+
+            if (property.PropertyType.FullName == HttpResponseType)
+            {
+                AddHttpOutputBinding(bindingMetadata);
+            }
+        }
+
+        private bool TryAddOutputBindingFromMethod(IList<ExpandoObject> bindingMetadata, MethodDefinition method)
+        {
+            bool foundBinding = false;
+
             foreach (CustomAttribute methodAttribute in method.CustomAttributes)
             {
                 if (IsOutputBindingType(methodAttribute))
                 {
-                    AddOutputBindingMetadata(bindingMetadata, methodAttribute);
+                    if (foundBinding)
+                    {
+                        // TODO: Is this an exception? If so, throw with proper message
+                        throw new Exception($"Found multiple Output bindings on method '{method.FullName}'. " +
```

Comment:
The model I had in mind (from our discussion, and samples) is that if you have multiple outputs, they need to come from the property of  the return type, and can't be on the method. Because we said that we don't want to add `Name` property yet, and there would be no way of mapping which output attribute on method corresponds to which binding name.